027-kubernetes-config-provider.md

```diff
@@ -0,0 +1,82 @@
+# Kubernetes Configuration Provider for Apache Kafka
+
+This proposal suggests to create a configuration provider for reading data from Kubernetes Secrets and Config Maps.
+
+## Current situation
+
+Apache Kafka supports pluggable configuration providers which can load configuration data from external sources.
+They can be used in the configuration of the different components.
+By default, Kafka provides two configuration providers:
+
+* FileConfigProvider for reading configuration records from properties files
+* DirectoryConfigProvider for reading configuration from complete files
+
+In addition, users can implement their own configuration providers as well.
+
+## Motivation
+
+Strimzi is using Kubernetes Secrets to store different information.
+For example cluster or user certificates or user passwords.
+Kafka clients (or other components) running in the same namespace can mount the secrets as volumes or environment variables and use them.
+But this does not work for applications running in other namespaces or outside the Kubernetes cluster.
+They have to either copy the secrets into their namespace or extract the files to use them.
+Users also need to make sure that their data are kept in sync and update them when the original secrets change.
+
+Having a configuration provider which can extract these data directly from the Kubernetes API would eliminate these issues.
+It would allow to load the configuration data from Kubernetes API even from other namespaces or outside of the Kubernetes cluster.
```

Comment:
```suggestion
It would allow users to load the configuration data from the Kubernetes API, even from other namespaces or outside of the Kubernetes cluster.
```